src/datanode/src/server/grpc.rs

```diff
@@ -104,6 +126,14 @@ impl Instance {
             .execute(SqlRequest::FlushTable(req), QueryContext::arc())
             .await
     }
+
+    async fn invalidate_catalog_cache_if_need(&self, catalog: &str, schema: &str, table: &str) {
```

Comment:
I think this function or these invalidations should be moved into the remote catalog manager. Then if other places alter or drop tables, the cache can be invalidated too.